opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -416,7 +417,9 @@ class Tracer:
     # This is the default behavior when creating spans.
     CURRENT_SPAN = Span()
 
-    def get_current_span(self) -> "Span":
+    def get_current_span(
+        self, context: typing.Optional[Context] = None
```

Comment:
As I understand it, returning the context here (and accepting it as an arg below) is required for "explicit" context propagation. But I still don't understand the use case for maintaining multiple contexts, and since the rest of the code assumes implicit (or "automatic" as per the spec) context propagation, I suspect there are more changes we'd have to make to make this work.